.github/workflows/build.yml

```diff
@@ -57,4 +65,26 @@ jobs:
         run: cargo hack doc --verbose --no-deps --each-feature --no-dev-deps --optional-deps url -p rumqttc -p rumqttd
 
       - name: Test rumqttc and rumqttd
-        run: cargo hack test --verbose --each-feature --optional-deps url -p rumqttc -p rumqttd
+        run: |
+            cargo hack --each-feature --optional-deps url llvm-cov -p rumqttc -p rumqttd --no-report nextest
```

Comment:
can you elaborate this part? is it running `cargo hack llvm-cov` and `llvm-con` runs `nextest`? 

if is kinda confusing to figure out what's exactly going on due to arrangement of arguments and flags, can we refactor if possible?

Reply:
so basically like:
```sh
cargo hack llvm-cov nextest --no-report --each-feature --optional-deps url -p rumqttc -p rumqttd
```

Reply:
> `--each-feature`

This is a flag of `cargo hack`, not `llvm-cov` or `nextest`

> `--optional-deps url`

This needs to be supplied to `cargo hack` to tell it that we need it include it as the optional dep when running test for each feature.

Reply:
can you please try it out? it works for me :)

Reply:
Since `--each-feature` is a flag of `cargo hack` not `llvm-cov` or `nextest`, do you see a problem with current approach?

Reply:
> For you.
> 
right :smile: readability is always dependent on person reading it, if i wrote it, I will find it readable as I know what I was doing, but that might not be the same for contributors / other maintainers of repo.

so, I don't wanna argue about which one is more readable. 

As a reviewer it was my duty to review it and present my thoughts. Rest is up-to the contributor :)

Reply:
as said earlier, choose whatever you feel readable :)